bin/node/src/service.rs

```diff
@@ -33,6 +35,24 @@ type FullClient = sc_service::TFullClient<Block, RuntimeApi, AlephExecutor>;
 type FullBackend = sc_service::TFullBackend<Block>;
 type FullSelectChain = sc_consensus::LongestChain<FullBackend, Block>;
 
+struct LimitNonfinalized(u32);
+
+impl<N: BaseArithmetic> BackoffAuthoringBlocksStrategy<N> for LimitNonfinalized {
+    fn should_backoff(
+        &self,
+        chain_head_number: N,
+        _chain_head_slot: Slot,
+        finalized_number: N,
+        _slow_now: Slot,
+        _logging_target: &str,
+    ) -> bool {
+        let nonfinalized_blocks: u32 = chain_head_number
+            .saturating_sub(finalized_number)
+            .unique_saturated_into();
+        nonfinalized_blocks >= self.0
```

Comment:
Can we log something when returning `true`? Ideally in `info` level.

Reply:
Good idea, added as `warn`, this shouldn't be happening during normal operation anyway.